sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -578,17 +588,44 @@ def __init__(self, value_coder, timestamp_coder, window_coder):
 
   def encode_to_stream(self, value, out, nested):
     wv = value  # type cast
-    self._value_coder.encode_to_stream(wv.value, out, True)
     # Avoid creation of Timestamp object.
-    out.write_bigendian_int64(wv.timestamp_micros)
+    restore_sign = -1 if wv.timestamp_micros < 0 else 1
+    out.write_bigendian_uint64(
+        # Convert to postive number and divide, since python rounds off to the
+        # lower negative number. For ex: -3 / 2 = -2, but we expect it to be -1,
+        # to be consistent across SDKs.
+        self._from_normal_time(
+            restore_sign * (abs(wv.timestamp_micros) / 1000)))
```

Comment:
Much better! Any performance implications dividing by float and converting to int? (I think not)